### PR TITLE
build(client): Update build-tools deps to 0.40.0

### PR DESCRIPTION
**Pull request by tylerbutler**
Updated the following:

  client (release group)

Dependencies on build-tools updated:

  @fluid-tools/build-cli: 0.40.0
  @fluidframework/build-tools: 0.40.0
  @fluidframework/bundle-size-tools: 0.40.0
  @fluid-tools/version-tools: 0.40.0

## Automated changes

This update caused two major changes:

- type tests are now more compact and several bugs have been addressed.
- Some api reports have comment-only changes.

## Manual changes

I also noticed that odsp-doclib-utils had a lint config that did not extend from "prettier," so some lint rules that shouldn't be running were getting triggered on the type tests. Extending the prettier config resolved the problem.

Finally, I added an exclusion of the npm-metadata-and-sorting policy for the root package.json because of [AB#8640](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8640), which I'll resolve later.

### CODE REVIEW

**Review by alexvy86 — APPROVED** [review]
Devtools looks good. I also looked at the root package.json and the fluid build config file.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -17225,8 +17225,8 @@ packages:
     dependencies:
       '@aws-crypto/sha256-browser': 3.0.0
       '@aws-crypto/sha256-js': 3.0.0
-      '@aws-sdk/client-sso-oidc': 3.569.0(@aws-sdk/client-sts@3.569.0)
-      '@aws-sdk/client-sts': 3.569.0
+      '@aws-sdk/client-sso-oidc': 3.569.0
+      '@aws-sdk/client-sts': 3.569.0(@aws-sdk/client-sso-oidc@3.569.0)
```
Comment (jason-ha) [inline pnpm-lock.yaml:17229]: Why oy why? Just revert these 11 collateral changes? Or are these the right changes?

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -32532,22 +32532,6 @@ packages:
...
-      typescript:
-        optional: true
-    dependencies:
-      import-fresh: 3.3.0
-      js-yaml: 4.1.0
-      parse-json: 5.2.0
-      path-type: 4.0.0
-      typescript: 5.1.6
-    dev: true
-
```
Comment (jason-ha) [inline pnpm-lock.yaml:32550]: I like these

**Review by jason-ha — APPROVED** [review]
Looked at package.json, osdp eslint config, lock file, and *.api.md files.

File: azure/packages/azure-service-utils/api-report/azure-service-utils.legacy.alpha.api.md [hunk 3]
```diff
@@ -11,6 +11,4 @@ export { IUser }
 
 export { ScopeType }
 
-// (No @packageDocumentation comment for this package)
```
Comment (CraigMacomber) [inline azure/packages/azure-service-utils/api-report/azure-service-utils.legacy.alpha.api.md:14]: Do you know why this went away?

**Review by CraigMacomber — APPROVED** [review]
Most of the changes are type tests related to my work, which seem to be working as expected.

File: azure/packages/azure-service-utils/api-report/azure-service-utils.legacy.alpha.api.md (hunk 3 above)
Comment (Josmithr) [reply azure/packages/azure-service-utils/api-report/azure-service-utils.legacy.alpha.api.md:14]: Jason added support for retaining the `@packageDocumentation` comments with our custom rollup generation.

**Review by Josmithr — APPROVED** [review]
Reviewed updated API reports

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**